rust-runtime/aws-smithy-runtime-api/src/client/interceptors/error.rs

```diff
@@ -14,6 +14,7 @@ pub type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
 #[derive(Debug)]
 pub struct InterceptorError {
     kind: ErrorKind,
+    context: Option<String>,
```

Comment:
Can/should this be a `Option<Cow<'static, str>>`?

Reply:
I'm not happy with the `InterceptorError` in its current incantation. Going to do some refactoring here.